### DISCUSSION TITLE
Add a [Pluto TV], [XUMO], [Roku], [Redbox] tag

### DISCUSSION THREAD

**Discussion by BellezaEmporium** [iptv-org/community#265] [Ideas]
This will permit to classify correctly the channel by their name and not by the origin of their streams.

Right now there's a bit of everything sadly, and we do not have a conventional way.

**Comment by Dum4G** [comment]
Some Redbox channels are coming from XUMO. Rather than using tags we should start channel names from it's origin name ex. Pluto Tv Drama, just like we do now

**Reply by BellezaEmporium** [reply]
Wouldn't it make a mess when it comes to searching certain channels that exists on all 3 servers ?

Pluto TV does host certain XUMO channels, and if we add as an example Pluto TV All Weddings and XUMO All Weddings, one would be classified way down compared to another.

Wouldn't it be simpler to do All Weddings [Pluto TV] and All Weddings [XUMO] ? In that way, we keep both channels into the correct order.

**Reply by Tribalize** [reply]
is the host going to be listed in the channel name after group-title,    
I can start that implemtation when i get time

**Comment by jkemp814** [comment]
STIRR also

**Comment by jkemp814** [comment]
I noticed there are original channels ABC, CBS, NBC coming from sources like Pluto, Redbox, XUMO, and STIRR, but they also have channels that they create. i.e. Pluto TV Thrriler, Pluto TV Westerns.

**Comment by Dum4G** [comment]
Origins were prioritized for a reason. Why would someone care for the source channels come from if it's not something exclusive? My suggestion is to keep them before the channel name without brackets as long as the channel (content) is exclusive to a source. If it's available everywhere let's just treat them as a single channel with multiple sources and without any suffixes/prefixes. All Weddings is a good example of such a case

**Reply by BellezaEmporium** [reply]
Well, as I've said it's to better identify the channels.

Of course if the channel is made by Pluto TV we'll keep Pluto TV in front of it, but for the rest it's like a mash of tons of different links and no conventions when it comes to identifications.

If I follow your reasoning, then the channels destined for French people, made by Pluto.tv should be on the FR playlist and not stay on the US one.

**Reply by Dum4G** [reply]
>but for the rest it's like a mash of tons of different links and no conventions when it comes to identifications

Yes? End viewer doesn't care what the source is if it's generally available channel. In fact, it's only the first link that is in genereated playlists that matters.

**Reply by BellezaEmporium** [reply]
Not only talking for the end viewer, but for us too. Would you be confused if the link you were reviewing also existed 300 lines below because instead of "Pluto TV Magic" it was saying "Magic (Pluto TV)" ?

**Reply by Dum4G** [reply]
1. In that case it would be
Magic (Pluto TV)
Magic (STIRR)
Magic  (XUMO)
2. Unlike viewers we can always press Ctrl+F

**Reply by BellezaEmporium** [reply]
> 
> 
>     1. In that case it would be
>        Magic (Pluto TV)
>        Magic (STIRR)
>        Magic  (XUMO)
> 
>     2. Unlike viewers we can always press Ctrl+F

1) That's how I would see things too. It would make things a lot clearer for us, and keep the playlists more maintainable.
2) If they're using VLC they also can, when you think about it. But no one would scroll over a few matches.

**Reply by Dum4G** [reply]
> 1. That's how I would see things too

Except I see them as
Magic
Magic
Magic

unless they're streaming different content like CNN from Pluto is different from other streams. Any channel that is exclusive to a platform remains as it is now

**Reply by jkemp814** [reply]
It needs to say something about the source  because CNN on Pluto TV is not live, but simply an older rebroadcast of segments and stories told on CNN.

**Reply by jkemp814** [reply]
Wow, you already said that, lol. It popped up as I hit send

**Comment by jkemp814** [comment]
I did a search on the us.m3u playlist and notice there are many channels with https://service-stitcher.clusters.pluto.tv/ in the link. (1,047) 

Does that mean those are really Pluto TV channels, even though that's the only mention of it being a Pluto TV channel? 

Should Pluto TV be added to the name of those channels? 

Out of that 1,047 results, I would say about 97 percent have Pluto TV at the front of the name and description.

Should those remaining channels have the same? (that have no mention of it really being from that source)

Some of those remaining channels do have Pluto TV at the end of channel description with in () but that is very few compared to the majority. (36)

**Reply by jkemp814** [reply]
So, Crackle, Fubu, Redbox, and Roku, are really XUMO?
 When looking at those channel links they seem to start off with:

https://dai2.xumo.com/
https://dai2-xumo.fubo.tv/
http://crackle-xumo.amagi.tv/
https://dai2.xumo.com/amagi_hls_data_xumo1212A-redbox...
https://dai2.xumo.com/amagi_hls_data_xumo1212A-roku...

**Reply by BellezaEmporium** [reply]
I think that some channels were on Pluto TV but were slowly taken off while not being deleted from their servers.

You're safe to do what you're doing with those links.

Yes, lots of channels are from the XUMO servers.

**Comment by freearhey** [comment]
This can now be done through Feed ID. For example:

```
Magic.us@PlutoTV
Magic.us@STIRR
Magic.us@XUMO
```
